node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/MetricsResponse.java

```diff
@@ -56,104 +57,120 @@ private MetricsResponse(Slime response, NodeList applicationNodes, NodeRepositor
-            double convertValue(double metricValue) { return (float)metricValue / 100; } // % to ratio
+            public List<String> metricResponseNames() { return List.of("mem.util"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).average().orElse(0) / 100; // % to ratio
+            }
         },
         disk { // a node resource
-            public String metricResponseName() { return "disk.util"; }
-            double convertValue(double metricValue) { return (float)metricValue / 100; } // % to ratio
+            public List<String> metricResponseNames() { return List.of("disk.util"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).average().orElse(0) / 100; // % to ratio
+            }
         },
         generation { // application config generation active on the node
-            public String metricResponseName() { return "application_generation"; }
-            double convertValue(double metricValue) { return (float)metricValue; } // Really a long
-            double defaultValue() { return -1.0; }
+            public List<String> metricResponseNames() { return List.of("application_generation"); }
+            double computeFinal(List<Double> values) {
+                return values.stream().mapToDouble(v -> v).min().orElse(-1);
+            }
         },
         inService {
-            public String metricResponseName() { return "in_service"; }
-            double convertValue(double metricValue) { return (float)metricValue; } // Really a boolean
-            double defaultValue() { return 1.0; }
+            public List<String> metricResponseNames() { return List.of("in_service"); }
+            double computeFinal(List<Double> values) {
+                // Really a boolean. Default true. If any is oos -> oos.
+                return values.stream().anyMatch(v -> v == 0) ? 0 : 1;
+            }
         },
         queryRate { // queries per second
-            public String metricResponseName() { return "queries.rate"; }
-            double convertValue(double metricValue) { return (float)metricValue; }
-            double defaultValue() { return 0.0; }
+            public List<String> metricResponseNames() {
+                return List.of("queries.rate",
+                               "content.proton.documentdb.matching.queries.rate");
+            }
         },
         writeRate { // writes per second
-            public String metricResponseName() { return "feed.http-requests.rate"; }
-            double convertValue(double metricValue) { return (float) metricValue; }
-            double defaultValue() { return 0.0; }
+            public List<String> metricResponseNames() {
+                return List.of("feed.http-requests.rate",
+                               "vds.filestor.alldisks.allthreads.put.sum.count.rate",
```

Comment:
Should these be added to `AutoscalingMetrics.java`?

Reply:
Yes, i forgot - thanks!